File: mRemoteV1/CS/Forms/OptionsPages/UpdatesPage.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.ComponentModel;
using mRemoteNG.My;
using mRemoteNG.App;
//using mRemoteNG.App.Info;
using mRemoteNG.Security;
using mRemoteNG.Tools;
using mRemoteNG.UI.Window;
using PSTaskDialog;


namespace mRemoteNG.Forms.OptionsPages
{
	public partial class UpdatesPage
		{
			public UpdatesPage()
			{
				InitializeComponent();
			}
#region Public Methods
public override string PageName
			{
				get
				{
					return Language.strTabUpdates;
				}
				set
				{
				}
			}
			
			public override void ApplyLanguage()
			{
				base.ApplyLanguage();
				
#if !PORTABLE
				lblUpdatesExplanation.Text = Language.strUpdateCheck;
#else
				lblUpdatesExplanation.Text = Language.strUpdateCheckPortableEdition;
#endif
				
				chkCheckForUpdatesOnStartup.Text = Language.strCheckForUpdatesOnStartup;
				btnUpdateCheckNow.Text = Language.strCheckNow;
				
				chkUseProxyForAutomaticUpdates.Text = Language.strCheckboxUpdateUseProxy;
				lblProxyAddress.Text = Language.strLabelAddress;
				lblProxyPort.Text = Language.strLabelPort;
				
				chkUseProxyAuthentication.Text = Language.strCheckboxProxyAuthentication;
				lblProxyUsername.Text = Language.strLabelUsername;
				lblProxyPassword.Text = Language.strLabelPassword;
				
				btnTestProxy.Text = Language.strButtonTestProxy;
			}
			
			public override void LoadSettings()
			{
				base.SaveSettings();
				
				chkCheckForUpdatesOnStartup.Checked = System.Convert.ToBoolean(My.Settings.Default.CheckForUpdatesOnStartup);
				cboUpdateCheckFrequency.Enabled = chkCheckForUpdatesOnStartup.Checked;
				cboUpdateCheckFrequency.Items.Clear();
				int nDaily = cboUpdateCheckFrequency.Items.Add(Language.strUpdateFrequencyDaily);
				int nWeekly = cboUpdateCheckFrequency.Items.Add(Language.strUpdateFrequencyWeekly);
				int nMonthly = cboUpdateCheckFrequency.Items.Add(Language.strUpdateFrequencyMonthly);
				if (My.Settings.Default.CheckForUpdatesFrequencyDays < 1)
				{
					chkCheckForUpdatesOnStartup.Checked = false;
					cboUpdateCheckFrequency.SelectedIndex = nDaily;
				} // Daily
				else if ((int) My.Settings.Default.CheckForUpdatesFrequencyDays == 1)
				{
					cboUpdateCheckFrequency.SelectedIndex = nDaily;
				} // Weekly
				else if ((int) My.Settings.Default.CheckForUpdatesFrequencyDays == 7)
				{
					cboUpdateCheckFrequency.SelectedIndex = nWeekly;
				} // Monthly
				else if ((int) My.Settings.Default.CheckForUpdatesFrequencyDays == 31)
				{
					cboUpdateCheckFrequency.SelectedIndex = nMonthly;
				}
				else
				{
					int nCustom = cboUpdateCheckFrequency.Items.Add(string.Format(Language.strUpdateFrequencyCustom, My.Settings.Default.CheckForUpdatesFrequencyDays));
					cboUpdateCheckFrequency.SelectedIndex = nCustom;
				}
				
				chkUseProxyForAutomaticUpdates.Checked = System.Convert.ToBoolean(My.Settings.Default.UpdateUseProxy);
				pnlProxyBasic.Enabled = System.Convert.ToBoolean(My.Settings.Default.UpdateUseProxy);
				txtProxyAddress.Text = System.Convert.ToString(My.Settings.Default.UpdateProxyAddress);
				numProxyPort.Value = System.Convert.ToDecimal(My.Settings.Default.UpdateProxyPort);
				
				chkUseProxyAuthentication.Checked = System.Convert.ToBoolean(My.Settings.Default.UpdateProxyUseAuthentication);
				pnlProxyAuthentication.Enabled = System.Convert.ToBoolean(My.Settings.Default.UpdateProxyUseAuthentication);
				txtProxyUsername.Text = System.Convert.ToString(My.Settings.Default.UpdateProxyAuthUser);
				txtProxyPassword.Text = Crypt.Decrypt(System.Convert.ToString(My.Settings.Default.UpdateProxyAuthPass), App.Info.General.EncryptionKey);
				
				btnTestProxy.Enabled = System.Convert.ToBoolean(My.Settings.Default.UpdateUseProxy);
				
#if PORTABLE
				foreach (Control Control in Controls)
				{
					if (Control != lblUpdatesExplanation)
					{
						Control.Visible = false;
					}
				}
#endif
			}
			
			public override void SaveSettings()
			{
				base.SaveSettings();
				
				My.Settings.Default.CheckForUpdatesOnStartup = chkCheckForUpdatesOnStartup.Checked;
				if (cboUpdateCheckFrequency.SelectedItem.ToString() == Language.strUpdateFrequencyDaily)
				{
					My.Settings.Default.CheckForUpdatesFrequencyDays = 1;
				}
				else if (cboUpdateCheckFrequency.SelectedItem.ToString() == Language.strUpdateFrequencyWeekly)
				{
					My.Settings.Default.CheckForUpdatesFrequencyDays = 7;
				}
				else if (cboUpdateCheckFrequency.SelectedItem.ToString() == Language.strUpdateFrequencyMonthly)
				{
					My.Settings.Default.CheckForUpdatesFrequencyDays = 31;
				}
				
				My.Settings.Default.UpdateUseProxy = chkUseProxyForAutomaticUpdates.Checked;
				My.Settings.Default.UpdateProxyAddress = txtProxyAddress.Text;
				My.Settings.Default.UpdateProxyPort = numProxyPort.Value;
				
				My.Settings.Default.UpdateProxyUseAuthentication = chkUseProxyAuthentication.Checked;
				My.Settings.Default.UpdateProxyAuthUser = txtProxyUsername.Text;
				My.Settings.Default.UpdateProxyAuthPass = Crypt.Encrypt(txtProxyPassword.Text, App.Info.General.EncryptionKey);
			}
#endregion
			
#region Private Fields
			private App.Update _appUpdate;
#endregion
			
#region Private Methods
#region Event Handlers
			public void chkCheckForUpdatesOnStartup_CheckedChanged(object sender, EventArgs e)
			{
				cboUpdateCheckFrequency.Enabled = chkCheckForUpdatesOnStartup.Checked;
			}
			
			public void btnUpdateCheckNow_Click(object sender, EventArgs e)
			{
				Runtime.Windows.Show(Type.Update);
			}
			
			public void chkUseProxyForAutomaticUpdates_CheckedChanged(object sender, EventArgs e)
			{
				pnlProxyBasic.Enabled = chkUseProxyForAutomaticUpdates.Checked;
				btnTestProxy.Enabled = chkUseProxyForAutomaticUpdates.Checked;
				
				if (chkUseProxyForAutomaticUpdates.Checked)
				{
					chkUseProxyAuthentication.Enabled = true;
					
					if (chkUseProxyAuthentication.Checked)
					{
						pnlProxyAuthentication.Enabled = true;
					}
				}
				else
				{
					chkUseProxyAuthentication.Enabled = false;
					pnlProxyAuthentication.Enabled = false;
				}
			}
			
			public void btnTestProxy_Click(object sender, EventArgs e)
			{
				if (_appUpdate != null)
				{
					if (_appUpdate.IsGetUpdateInfoRunning)
					{
						return ;
					}
				}
				
				_appUpdate = new App.Update();
				_appUpdate.Load += _appUpdate.Update_Load;
				_appUpdate.SetProxySettings(chkUseProxyForAutomaticUpdates.Checked, txtProxyAddress.Text, (int) numProxyPort.Value, chkUseProxyAuthentication.Checked, txtProxyUsername.Text, txtProxyPassword.Text);
				
				btnTestProxy.Enabled = false;
				btnTestProxy.Text = Language.strOptionsProxyTesting;
				
				_appUpdate.GetUpdateInfoCompletedEvent += GetUpdateInfoCompleted;
				
				_appUpdate.GetUpdateInfoAsync();
			}
			
			public void chkUseProxyAuthentication_CheckedChanged(object sender, EventArgs e)
			{
				if (chkUseProxyForAutomaticUpdates.Checked)
				{
					if (chkUseProxyAuthentication.Checked)
					{
						pnlProxyAuthentication.Enabled = true;
					}
					else
					{
						pnlProxyAuthentication.Enabled = false;
					}
				}
			}
#endregion
			
			private void GetUpdateInfoCompleted(object sender, AsyncCompletedEventArgs e)
			{
				if (InvokeRequired)
				{
					AsyncCompletedEventHandler myDelegate = new AsyncCompletedEventHandler(GetUpdateInfoCompleted);
					Invoke(myDelegate, new object[] {sender, e});
					return ;
				}
				
				try
				{
					_appUpdate.GetUpdateInfoCompletedEvent -= GetUpdateInfoCompleted;
					
					btnTestProxy.Enabled = true;
					btnTestProxy.Text = Language.strButtonTestProxy;
					
					if (e.Cancelled)
					{
						return ;
					}
					if (e.Error != null)
					{
						throw (e.Error);
					}
					
					cTaskDialog.ShowCommandBox(this, System.Convert.ToString(Application.Info.ProductName), Language.strProxyTestSucceeded, "", Language.strButtonOK, false);
				}
				catch (Exception ex)
				{
					cTaskDialog.ShowCommandBox(this, System.Convert.ToString(Application.Info.ProductName), Language.strProxyTestFailed, Misc.GetExceptionMessageRecursive(ex), "", "", "", Language.strButtonOK, false, eSysIcons.Error, (PSTaskDialog.eSysIcons) 0);
				}
			}
#endregion
		}
}

File: mRemoteV1/Forms/OptionsPages/UpdatesPage.vb
Imports System.ComponentModel
Imports mRemote3G.App
Imports mRemote3G.App.Info
Imports mRemote3G.Security
Imports mRemote3G.Tools
Imports mRemote3G.UI.Window
Imports PSTaskDialog

Namespace Forms.OptionsPages
    Public Class UpdatesPage

#Region "Public Methods"

        Public Overrides Property PageName As String
            Get
                Return Language.Language.strTabUpdates
            End Get
            Set
            End Set
        End Property

        Public Overrides Sub ApplyLanguage()
            ' Temporary
            Return
            MyBase.ApplyLanguage()

#If Not PORTABLE Then
            lblUpdatesExplanation.Text = Language.Language.strUpdateCheck
#Else
            lblUpdatesExplanation.Text = Language.strUpdateCheckPortableEdition
#End If

            chkCheckForUpdatesOnStartup.Text = Language.Language.strCheckForUpdatesOnStartup
            btnUpdateCheckNow.Text = Language.Language.strCheckNow

            chkUseProxyForAutomaticUpdates.Text = Language.Language.strCheckboxUpdateUseProxy
            lblProxyAddress.Text = Language.Language.strLabelAddress
            lblProxyPort.Text = Language.Language.strLabelPort

            chkUseProxyAuthentication.Text = Language.Language.strCheckboxProxyAuthentication
            lblProxyUsername.Text = Language.Language.strLabelUsername
            lblProxyPassword.Text = Language.Language.strLabelPassword

            btnTestProxy.Text = Language.Language.strButtonTestProxy
        End Sub

        Public Overrides Sub LoadSettings()
            ' Temporary
            Return
            MyBase.SaveSettings()

            chkCheckForUpdatesOnStartup.Checked = My.Settings.CheckForUpdatesOnStartup
            cboUpdateCheckFrequency.Enabled = chkCheckForUpdatesOnStartup.Checked
            cboUpdateCheckFrequency.Items.Clear()
            Dim nDaily As Integer = cboUpdateCheckFrequency.Items.Add(Language.Language.strUpdateFrequencyDaily)
            Dim nWeekly As Integer = cboUpdateCheckFrequency.Items.Add(Language.Language.strUpdateFrequencyWeekly)
            Dim nMonthly As Integer = cboUpdateCheckFrequency.Items.Add(Language.Language.strUpdateFrequencyMonthly)
            Select Case My.Settings.CheckForUpdatesFrequencyDays
                Case Is < 1
                    chkCheckForUpdatesOnStartup.Checked = False
                    cboUpdateCheckFrequency.SelectedIndex = nDaily
                Case 1 ' Daily
                    cboUpdateCheckFrequency.SelectedIndex = nDaily
                Case 7 ' Weekly
                    cboUpdateCheckFrequency.SelectedIndex = nWeekly
                Case 31 ' Monthly
                    cboUpdateCheckFrequency.SelectedIndex = nMonthly
                Case Else
                    Dim nCustom As Integer =
                            cboUpdateCheckFrequency.Items.Add(String.Format(Language.Language.strUpdateFrequencyCustom,
                                                                            My.Settings.CheckForUpdatesFrequencyDays))
                    cboUpdateCheckFrequency.SelectedIndex = nCustom
            End Select

            chkUseProxyForAutomaticUpdates.Checked = My.Settings.UpdateUseProxy
            pnlProxyBasic.Enabled = My.Settings.UpdateUseProxy
            txtProxyAddress.Text = My.Settings.UpdateProxyAddress
            numProxyPort.Value = My.Settings.UpdateProxyPort

            chkUseProxyAuthentication.Checked = My.Settings.UpdateProxyUseAuthentication
            pnlProxyAuthentication.Enabled = My.Settings.UpdateProxyUseAuthentication
            txtProxyUsername.Text = My.Settings.UpdateProxyAuthUser
            txtProxyPassword.Text = Crypt.Decrypt(My.Settings.UpdateProxyAuthPass, General.EncryptionKey)

            btnTestProxy.Enabled = My.Settings.UpdateUseProxy

#If PORTABLE Then
        For Each Control As Control In Controls
            If Control IsNot lblUpdatesExplanation Then
                Control.Visible = False
            End If
        Next
#End If
        End Sub

        Public Overrides Sub SaveSettings()
            ' Temporary
            Return
            MyBase.SaveSettings()

            My.Settings.CheckForUpdatesOnStartup = chkCheckForUpdatesOnStartup.Checked
            Select Case cboUpdateCheckFrequency.SelectedItem.ToString()
                Case Language.Language.strUpdateFrequencyDaily
                    My.Settings.CheckForUpdatesFrequencyDays = 1
                Case Language.Language.strUpdateFrequencyWeekly
                    My.Settings.CheckForUpdatesFrequencyDays = 7
                Case Language.Language.strUpdateFrequencyMonthly
                    My.Settings.CheckForUpdatesFrequencyDays = 31
            End Select

            My.Settings.UpdateUseProxy = chkUseProxyForAutomaticUpdates.Checked
            My.Settings.UpdateProxyAddress = txtProxyAddress.Text
            My.Settings.UpdateProxyPort = numProxyPort.Value

            My.Settings.UpdateProxyUseAuthentication = chkUseProxyAuthentication.Checked
            My.Settings.UpdateProxyAuthUser = txtProxyUsername.Text
            My.Settings.UpdateProxyAuthPass = Crypt.Encrypt(txtProxyPassword.Text, General.EncryptionKey)
        End Sub

#End Region

#Region "Private Fields"

        Private _appUpdate As App.Update

#End Region

#Region "Private Methods"

#Region "Event Handlers"

        Private Sub chkCheckForUpdatesOnStartup_CheckedChanged(sender As Object, e As EventArgs) _
            Handles chkCheckForUpdatesOnStartup.CheckedChanged
            ' Temporary
            Return
            cboUpdateCheckFrequency.Enabled = chkCheckForUpdatesOnStartup.Checked
        End Sub

        Private Sub btnUpdateCheckNow_Click(sender As Object, e As EventArgs) Handles btnUpdateCheckNow.Click
            ' Temporary
            Return
            Runtime.Windows.Show(Type.Update)
        End Sub

        Private Sub chkUseProxyForAutomaticUpdates_CheckedChanged(sender As Object, e As EventArgs) _
            Handles chkUseProxyForAutomaticUpdates.CheckedChanged
            ' Temporary
            Return
            pnlProxyBasic.Enabled = chkUseProxyForAutomaticUpdates.Checked
            btnTestProxy.Enabled = chkUseProxyForAutomaticUpdates.Checked

            If chkUseProxyForAutomaticUpdates.Checked Then
                chkUseProxyAuthentication.Enabled = True

                If chkUseProxyAuthentication.Checked Then
                    pnlProxyAuthentication.Enabled = True
                End If
            Else
                chkUseProxyAuthentication.Enabled = False
                pnlProxyAuthentication.Enabled = False
            End If
        End Sub

        Private Sub btnTestProxy_Click(sender As Object, e As EventArgs) Handles btnTestProxy.Click
            ' Temporary
            Return
            If _appUpdate IsNot Nothing Then
                If _appUpdate.IsGetUpdateInfoRunning Then Return
            End If

            _appUpdate = New App.Update
            _appUpdate.SetProxySettings(chkUseProxyForAutomaticUpdates.Checked, txtProxyAddress.Text, numProxyPort.Value,
                                        chkUseProxyAuthentication.Checked, txtProxyUsername.Text, txtProxyPassword.Text)

            btnTestProxy.Enabled = False
            btnTestProxy.Text = Language.Language.strOptionsProxyTesting

            AddHandler _appUpdate.GetUpdateInfoCompletedEvent, AddressOf GetUpdateInfoCompleted

            _appUpdate.GetUpdateInfoAsync()
        End Sub

        Private Sub chkUseProxyAuthentication_CheckedChanged(sender As Object, e As EventArgs) _
            Handles chkUseProxyAuthentication.CheckedChanged
            ' Temporary
            Return
            If chkUseProxyForAutomaticUpdates.Checked Then
                If chkUseProxyAuthentication.Checked Then
                    pnlProxyAuthentication.Enabled = True
                Else
                    pnlProxyAuthentication.Enabled = False
                End If
            End If
        End Sub

#End Region

        Private Sub GetUpdateInfoCompleted(sender As Object, e As AsyncCompletedEventArgs)
            ' Temporary
            Return
            If InvokeRequired Then
                Dim myDelegate As New AsyncCompletedEventHandler(AddressOf GetUpdateInfoCompleted)
                Invoke(myDelegate, New Object() {sender, e})
                Return
            End If

            Try
                RemoveHandler _appUpdate.GetUpdateInfoCompletedEvent, AddressOf GetUpdateInfoCompleted

                btnTestProxy.Enabled = True
                btnTestProxy.Text = Language.Language.strButtonTestProxy

                If e.Cancelled Then Return
                If e.Error IsNot Nothing Then Throw e.Error

                cTaskDialog.ShowCommandBox(Me, Application.ProductName, Language.Language.strProxyTestSucceeded, "",
                                           Language.Language.strButtonOK, False)
            Catch ex As Exception
                cTaskDialog.ShowCommandBox(Me, Application.ProductName, Language.Language.strProxyTestFailed,
                                           Misc.GetExceptionMessageRecursive(ex), "", "", "",
                                           Language.Language.strButtonOK, False, eSysIcons.Error, 0)
            End Try
        End Sub

#End Region
    End Class
End Namespace